angel_system/data/medical/load_bbn_data.py

```diff
@@ -21,6 +22,29 @@
 os.environ["CUDA_VISIBLE_DEVICES"] = "0, 1, 2, 3"
 
 
+
+RE_FILENAME_TIME = re.compile(
+    r"(?P<task>\w+)-(?P<vid>\d+)_(?P<frame>\d+).(?P<ext>\w+)"
+)
+
+
+def time_from_name(fname):
+    """
+    Extract the float timestamp from the filename.
+
+    :param fname: Filename of an image in the format
+        frame_<frame number>_<seconds>_<nanoseconds>.<extension>
```

Comment:
Is there somewhere else that it would be useful to document this filename requirement?

Reply:
Another idea: wherever such a filename requirement is assumed, we could add an "assert" checking for the right number of underscores

Reply:
^From Hannah:

the id used to be parsed like this:
```
RE_FILENAME_TIME = re.compile(
    r"frame_(?P<frame>\d+)(?P<ts>\d+(?:|.)\d+).(?P<ext>\w+)"
)
```
So then you'd get a slightly-opaque error in this code:
```
    fname = os.path.basename(fname)
    match = RE_FILENAME_TIME.match(fname)
    time = match.group("ts")
```

Rough proposal: edit the function [here](https://github.com/PTG-Kitware/angel_system/blob/master/angel_system/data/common/load_data.py#L52) and always use it to acquire any arguments from the assumed filename above. Add some assertions (correct number of underscores, etc.) which try to enforce the above naming convention. If those assertions fail, the failure message should contain the `fname` convention that's documented in-line above.

@Purg , please tell me if you have better ideas or suggestions for the check I'm proposing here!

Reply:
@hdefazio

Reply:
Late to the party. I would probably recommend updating the regex to have more specific captures for the expected pattern instead of capturing a variable pattern and then performing secondary parsing. Unless you're implying with the regex that the `_<nanoseconds>` portion is optional and sometimes not there. If the latter is the case, I would still recommend being explicit and having *two* regexs, one with and one without nanoseconds. Try matching against nanoseconds, if fail try parse against only seconds, if fail then raise an exception.
```python
RE_FILENAME_TIME_SEC = re.compile(
    r"frame_(?P<frame>\d+)_(?P<ts_sec>\d+)\.(?P<ext>\w+)"
)
RE_FILENAME_TIME_NANO = re.compile(
    r"frame_(?P<frame>\d+)_(?P<ts_sec>\d+)_(?P<ts_nano>\d+)\.(?P<ext>\w+)"
)
```